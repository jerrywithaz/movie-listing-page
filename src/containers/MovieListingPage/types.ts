import { Cinema, SelectedCinema, Session, Film } from "core/types";

export type MovieListingPageMappedProps = {
    selectedTheater: SelectedCinema,
    theaters: Cinema[],
    films: Film[]
};

export type MovieListingPageMappedDispatch = {
    setSelectedTheater: (cinema: Cinema) => void
};

export type MovieListingPageProps = (
    React.HTMLAttributes<HTMLDivElement> & 
    MovieListingPageMappedProps & 
    MovieListingPageMappedDispatch
);