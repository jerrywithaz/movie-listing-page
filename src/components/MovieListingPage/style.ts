import styled from 'styled-components';

export const MovieListingPage = styled.div`
    width: 100%;
    height: 100%;
    background: ${props => props.theme.palette.gray.primary};
`;