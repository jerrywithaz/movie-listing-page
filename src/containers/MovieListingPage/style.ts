import styled from 'styled-components';
import DefaultHero from 'components/Hero';
import DefaultFindAMovie from 'components/FindAMovie';

export const MovieListingPage = styled.div`
    width: 100%;
    height: 100%;
    background: ${props => props.theme.palette.gray.primary};
    display: flex;
    position: relative;
    align-items: center;
    justify-content: center;
`;

export const Hero = styled(DefaultHero)`
    position: absolute;
    width: 100%;
    top: 0;
    left: 0;
    z-index: 1;
`;

export const FindAMovie = styled(DefaultFindAMovie)`
    position: relative;
    z-index: 2;
`;