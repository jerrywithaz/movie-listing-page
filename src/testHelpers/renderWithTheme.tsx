import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;