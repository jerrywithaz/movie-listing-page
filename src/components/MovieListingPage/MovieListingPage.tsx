import React from 'react';

import * as Styled from './style';

function MovieListingPage() {
    return (
        <Styled.MovieListingPage>

        </Styled.MovieListingPage>
    );
}

export default MovieListingPage;