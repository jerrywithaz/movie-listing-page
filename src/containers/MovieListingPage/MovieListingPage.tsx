import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Dispatch } from 'redux';
import { makeSelectCinemas, makeSelectSelectedCinema } from 'core/selectors';
import { MovieListingPageProps, MovieListingPageMappedProps, MovieListingPageMappedDispatch } from './types';
import { AppState, Cinema } from 'core/types';
import { setSelectedCinema } from 'core/actions';

import * as Styled from './style';


function MovieListingPage({
    theaters, 
    selectedTheater,
    setSelectedTheater,
    ...rest
}: MovieListingPageProps) {

    return (
        <Styled.MovieListingPage {...rest}>
            <Styled.Hero/>
            <Styled.FindAMovie 
                theaters={theaters} 
                selectedTheater={selectedTheater}
                setSelectedTheater={setSelectedTheater}/>
        </Styled.MovieListingPage>
    );

}

const mapStateToProps = createStructuredSelector<AppState, MovieListingPageMappedProps>({
    theaters: makeSelectCinemas(),
    selectedTheater: makeSelectSelectedCinema()
});

const mapDispatchToProps = (dispatch: Dispatch): MovieListingPageMappedDispatch => {
    return {
        setSelectedTheater: (cinema: Cinema) => dispatch(setSelectedCinema(cinema))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieListingPage);