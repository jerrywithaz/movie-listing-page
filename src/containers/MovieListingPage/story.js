import React from 'react';
import { storiesOf } from '@storybook/react';
import MovieListingPage from '.';
import { getMockFilms, getMockTheaters } from 'core/mocks';

const story = storiesOf('MovieListingPage', module);

const films = getMockFilms();
const theaters = getMockTheaters();

story.add('Default', () => <MovieListingPage films={films} theaters={theaters}/>);
  