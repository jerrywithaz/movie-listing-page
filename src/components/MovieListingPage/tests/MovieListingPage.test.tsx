import React from 'react';
import MovieListingPage from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';

describe("components/MovieListingPage", () => {

    it("should match snapshot", () => {

        const tree = renderWithTheme(<MovieListingPage/>).toJSON();

        expect(tree).toMatchSnapshot();

    });

});