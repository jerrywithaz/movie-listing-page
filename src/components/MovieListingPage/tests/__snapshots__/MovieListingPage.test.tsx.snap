// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MovieListingPage should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background: #E5E5E5;
}

<div
  className="c0"
/>
`;
